test/e2e/ubernetes_lite.go

```diff
@@ -61,8 +64,136 @@ var _ = framework.KubeDescribe("Multi-AZ Clusters", func() {
 	It("should schedule pods in the same zones as statically provisioned PVs", func() {
 		PodsUseStaticPVsOrFail(f, (2*zoneCount)+1, image)
 	})
+
+	It("should only be allowed to provision PDs in zones where nodes exist", func() {
+		OnlyAllowNodeZones(f, zoneCount, image)
+	})
 })
 
+// OnlyAllowNodeZones tests that GetAllCurrentZones returns only zones with Nodes
+func OnlyAllowNodeZones(f *framework.Framework, zoneCount int, image string) {
+	gceCloud, err := framework.GetGCECloud()
```

Comment:
Do we need to add a check to make sure this test only runs for gce and gke?

Reply:
I believe thats already covered: https://github.com/davidz627/kubernetes/blob/38fd65b0cfdc84f47ee03c22ff0462a24be4c9ec/test/e2e/ubernetes_lite.go#L46

Reply:
unless you mean that I should have it not run on AWS. In which case that could be added.